model_analyzer/result/result_manager.py

```diff
@@ -108,20 +112,22 @@ def _create_server_table(self):
     def _create_inference_table(self):
         # Inference only
         inference_output_headers = []
-        inference_output_fields = self._config.inference_output_fields
-        for inference_output_field in inference_output_fields:
+        inference_output_fields = []
+        for inference_output_field in self._config.inference_output_fields:
             if inference_output_field in self.headers:
                 inference_output_headers.append(
                     self.headers[inference_output_field])
             elif inference_output_field in self._non_gpu_metrics_to_headers:
                 inference_output_headers.append(
                     self._non_gpu_metrics_to_headers[inference_output_field])
             else:
-                raise TritonModelAnalyzerException(
-                    f'Inference output field "{inference_output_field}", does not exist'
+                logging.warning(
```

Comment:
Can we modify the `server_output_fields` and `gpu_output_fields` based on the `metrics` config variable so that we don't have to ignore the headers in case they don't exist?

Reply:
This is tricky because the user can specify different metrics during `profile` and `analyze`. 

If we ask the user to input the `--metrics` flag during `analyze`, the flag is redundant with `server_output_fields`, `inference_output_fields` and `gpu_output_fields`. If we capture the `--metrics` flag during `profile`, it is not necessary since we are finding out exactly which metrics were collected.

The warning is due to a discrepancy between metrics during `profile` and `analyze`. I think it is a good idea we keep the default metrics during `analyze` phase a superset of `profile` phase, to avoid cases like the user includes `cpu_used_ram` during `profile` but forget to includes it during `analyze`.

Can we print the opposite of the warning? We inform the user there are metrics profiled but not analyzed, instead of there are not profiled metrics being analyzed.

Reply:
I think its best if analyze is able to automatically detect what metrics were profiled. Unfortunately this would likely make it necessary for analyze to find out the largest superset of metrics that were collected during profile. This is because it is possible to run a profile with metrics X,Y,Z then run another profile with metrics X,Y or Y,Z, or W,X,Y. When analyze is asked to export the data, it must somehow unify these.

Reply:
If the user is requesting metrics be in the output tables that were not collect, we must warn them and exclude those from the output.

Reply:
> I think its best if analyze is able to automatically detect what metrics were profiled. Unfortunately this would likely make it necessary for analyze to find out the largest superset of metrics that were collected during profile. This is because it is possible to run a profile with metrics X,Y,Z then run another profile with metrics X,Y or Y,Z, or W,X,Y. When analyze is asked to export the data, it must somehow unify these.

This is achieved by line 173-205.

Reply:
> If the user is requesting metrics be in the output tables that were not collected, we must warn them and exclude those from the output.

I will not update the behavior, but change the default. Then, the user should remember to update metrics in the `analyze` part if the `profile` part includes CPU metrics.